internal/controller/streams.go

```diff
@@ -353,6 +409,27 @@ func manageStream(logger *zap.SugaredLogger, connectAndStream func(*zap.SugaredL
 
 // ConnectStreams will continue to reboot and restart the main operations within the operator if any disconnects or errors occur.
 func ConnectStreams(ctx context.Context, logger *zap.SugaredLogger, envMap EnvironmentConfig, bufferedGrpcSyncer *BufferedGrpcWriteSyncer) {
+	// Falco channels communicate news events between http server and our network flows strea,
+	falcoEventChan := make(chan string)
+	http.HandleFunc("/", NewFalcoEventHandler(falcoEventChan))
+	falcoEvent := &http.Server{Addr: ":5000"}
+	// Start our falco server and have it passively listen, if it fails, try to just restart it.
+	go func() {
+		for {
+			func() {
+				defer func() {
+					if r := recover(); r != nil {
+						logger.Infow("Recovered from panic. Restarting server...", "error", r)
+					}
+				}()
+				if err := falcoEvent.ListenAndServe(); err != nil {
+					logger.Fatal("Falco server failed", zap.Error(err))
```

Comment:
`logger.Fatal` just exits the process. So what's the point of this loop and the `recover()`, etc.?
https://github.com/uber-go/zap/blob/966925341b9f43d835469b153e907cc9e1003c17/logger.go#L292-L293

Reply:
Yeah looking back (after 2 weeks) ill simplify this